operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -165,25 +168,58 @@ func extractAzureConfigSecret(s *corev1.Secret) (*storage.AzureStorageConfig, er
 	if len(container) == 0 {
 		return nil, fmt.Errorf("%w: %s", errSecretMissingField, storage.KeyAzureStorageContainerName)
 	}
-	name := s.Data[storage.KeyAzureStorageAccountName]
-	if len(name) == 0 {
-		return nil, fmt.Errorf("%w: %s", errSecretMissingField, storage.KeyAzureStorageAccountName)
-	}
-	key := s.Data[storage.KeyAzureStorageAccountKey]
-	if len(key) == 0 {
-		return nil, fmt.Errorf("%w: %s", errSecretMissingField, storage.KeyAzureStorageAccountKey)
+	workloadIdentity, err := validateAzureCredentials(s)
+	if err != nil {
+		return nil, err
 	}
 
 	// Extract and validate optional fields
 	endpointSuffix := s.Data[storage.KeyAzureStorageEndpointSuffix]
 
 	return &storage.AzureStorageConfig{
-		Env:            string(env),
-		Container:      string(container),
-		EndpointSuffix: string(endpointSuffix),
+		Env:              string(env),
+		Container:        string(container),
+		EndpointSuffix:   string(endpointSuffix),
+		WorkloadIdentity: workloadIdentity,
 	}, nil
 }
 
+func validateAzureCredentials(s *corev1.Secret) (workloadIdentity bool, err error) {
+	accountName := s.Data[storage.KeyAzureStorageAccountName]
+	accountKey := s.Data[storage.KeyAzureStorageAccountKey]
+	clientID := s.Data[storage.KeyAzureStorageClientID]
+	tenantID := s.Data[storage.KeyAzureStorageTenantID]
+	subscriptionID := s.Data[storage.KeyAzureStorageSubscriptionID]
+
+	if len(accountName) == 0 {
+		return false, fmt.Errorf("%w: %s", errSecretMissingField, storage.KeyAzureStorageAccountName)
+	}
```

Comment:
The validation does not look correct to me at this point. IIRC the only fields we should require for workload identity are: `environment`, `clientID`, `tenantID` and `subscriptionID`. If I am not mistaken we always require `accountName` here but it's not used by the azure SDK afaics.